updater/aws_test.go

```diff
@@ -1012,3 +1013,103 @@ func TestActivateInstance(t *testing.T) {
 		})
 	}
 }
+
+func TestAlreadyRunning(t *testing.T) {
+	cases := []struct {
+		name        string
+		listOut     *ecs.ListTasksOutput
+		listErr     error
+		expectedOk  bool
+		expectedErr string
+	}{
+		{
+			name: "success",
+			listOut: &ecs.ListTasksOutput{
+				TaskArns: []*string{
+					aws.String("tarsk-arn-1"),
+					aws.String("tarsk-arn-2"),
+				},
+			},
+			expectedOk: true,
+		},
+		{
+			name: "only one task",
+			listOut: &ecs.ListTasksOutput{
+				TaskArns: []*string{
+					aws.String("tarsk-arn-1"),
+				},
+			},
+			expectedOk: false,
+		},
+		{
+			name:        "fail list task",
+			listErr:     errors.New("failed to list task"),
+			expectedOk:  false,
+			expectedErr: "failed to list task",
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			mockECS := MockECS{
+				ListTasksFn: func(input *ecs.ListTasksInput) (*ecs.ListTasksOutput, error) {
+					return tc.listOut, tc.listErr
+				},
+			}
+			u := updater{ecs: mockECS, cluster: "ecs-cluster"}
+			ok, err := u.alreadyRunning("updater-family")
+			if tc.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), tc.expectedErr)
+			}
+			assert.Equal(t, tc.expectedOk, ok)
+		})
+	}
+}
+
+func TestTaskDefFamily(t *testing.T) {
+	cases := []struct {
+		name           string
+		taskDefARN     string
+		expectedErr    string
+		expectedFamily string
+	}{
+		{
+			name:           "success",
+			taskDefARN:     "arn:aws:ecs:us-west-2:1234567:task-definition/updater-family:1",
+			expectedFamily: "updater-family",
+		},
+		{
+			name:           "fail parse arn",
+			taskDefARN:     "arn:ecs:us-west-2:1234567updater-family:1",
+			expectedFamily: "",
+			expectedErr:    "arn: not enough sections",
+		},
+		{
+			name:           "fail empty arn",
+			taskDefARN:     "",
+			expectedFamily: "",
+			expectedErr:    "arn: invalid prefix",
+		},
+		{
+			name:           "fail extract family",
+			taskDefARN:     "arn:aws:ecs:us-west-2:1234567:task-def/updater-family1",
+			expectedFamily: "",
+			expectedErr:    "not a task definition arn:",
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			os.Setenv("TaskDefinitionARN", tc.taskDefARN)
```

Comment:
We should restore the original value when manipulating environment variables for tests.
```suggestion
			originalValue := os.Getenv(taskDefARNEnv)
			defer func() {os.Setenv(taskDefARNEnv, originalValue)}()
			os.Setenv(taskDefArnEnv, tc.taskDefARN)
```